helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -74,7 +74,7 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
       if (optimalAssignment.hasAnyFailure()) {
         String errorMessage = String.format(
             "Unable to find any available candidate node for partition %s; Fail reasons: %s",
-            replica.getPartitionName(), optimalAssignment.getFailures());
+            replica.getPartitionName(), optimalAssignment.getErrorMessage());
```

Comment:
OptimalAssignment is not an Exception.
The meaning of "ErrorMessage" is not clear.

I would prefer to call it AssignmentFailures or just keep it as it was.